polyfill/lib/ecmascript.mjs

```diff
@@ -175,9 +404,9 @@ export const ES = ObjectAssign(ObjectAssign(ObjectAssign({}, Cast), ES2019), {
   },
   GetTimeZoneDateTimeParts: (epochNanoseconds, timeZone) => {
     const offset = parseOffsetString(timeZone);
-
-    let epochMilliseconds = bigInt(epochNanoseconds).divide(1e6);
-    let nanos = +((epochNanoseconds < 0 ? 1e9 : 0) + bigInt(epochNanoseconds).mod(1e9));
+    let nanos = bigInt(epochNanoseconds).mod(1e9);
+    let epochMilliseconds = bigInt(epochNanoseconds).divide(1e9).multiply(1e3).plus(Math.floor(nanos / 1e6));
+    nanos = +((epochNanoseconds < 0 ? 1e9 : 0) + nanos);
```

Comment:
This is about the fact that big-integer returns numbers when it can and the fact that if you get the  parts from the `Intl.DateFormat` then the precision of that is *1 second* which is  a problem when you are trying to get deal with negative epochNanos and higher precision.

0n = 1970-01-01T00:00:00.000000000Z
-1n = 1969-12-31T23:59:59.999999999Z
But formater has 1 seocnd precision so just cuting off would result in
1970-01-01T00:00:00.000999999Z this makes sure we use the millisecond precision to ensure we stay below epoch even though only by micros/nanos.